_posts/2024-01-03-opensearch-performance-improvements.md

```diff
@@ -0,0 +1,1151 @@
+        <td>1%</td>
+    </tr>
+    <tr>
+        <td><code>range</code></td>
+        <td>74.5</td>
+        <td>77.5</td>
+        <td>-4%</td>
+    </tr>
+</table>
+
+#### Detailed results: Index data size on disk (bytes) with `zstd` compression using a single shard
+
+**Setup:** OpenSearch 2.11.0, single node (r5.xlarge) with 32 GB RAM and 16 GB heap. Index settings: 1 shard and 0 replicas.
+
+**Data size of disk results (bytes)**: The following table illustrates a benchmark comparison of the on-disk data size for the `http_logs` and `pmc` workloads for OpenSearch 2.11 with default vs. `zstd` codec enabled, including percentage improvement.
+
+<table>
+    <tr>
+        <th></th>
+        <th>Default compression (bytes)</th>
+        <th>ZSTD (bytes)</th>
+        <th>ZSTD_NO_DICT (bytes)</th>
+        <th>ZSTD improvement (vs. default codec)</th>
+        <th>ZSTD_NO_DICT improvement (vs. default codec)</th>
+    </tr>
+    <tr class="green-clr">
+        <td><code>http_logs</code></td>
+        <td>20,056,770,878.5</td>
+        <td>15,800,734,037</td>
+        <td>16,203,187,551</td>
+        <td>21%</td>
+        <td>19%</td>
+    </tr>
+    <tr class="green-clr">
+        <td><code>pmc</code></td>
+        <td>20,701,211,614.5</td>
+        <td>15,608,881,718.5</td>
+        <td>15,822,040,185</td>
+        <td>25%</td>
+        <td>24%</td>
+    </tr>
+</table>
+
+### Concurrent search improvements (*Experimental in 2.11*)
+
+OpenSearch users can now achieve better execution speed with concurrent segment search launched as experimental in OpenSearch 2.11. By default, OpenSearch processes a request sequentially across all the data segments on each shard during the query phase of a search request execution. With concurrent search, every shard-level request can concurrently search across segments during the query phase. Each shard divides its segments into [multiple slices](https://opensearch.org/blog/concurrent_segment_search/), where each slice serves as a unit of work executed in parallel on a separate thread. Therefore, the slice count governs the maximum degree of parallelism for a shard-level request. After all the slices finish their tasks, OpenSearch executes a reduce operation on the slices, merging them to generate the final result for the shard-level request.
+
+The benchmark results presented below show the benefits of concurrent search in action. These are the averages of data generated from over 4 days of nightly runs using OpenSearch Benchmark.
+
+#### Highlights
+
+Here are the key highlights:
+
+* An increase in performance with aggregate queries on workloads such as `nyc_taxis` workload, showcasing an improvement ranging between 50% to 70% over the default configuration.
+* The log analytics use cases for range queries demonstrated an improvement of around 65%.
+* Aggregation queries with hourly data aggregations, such as those for the `http_logs` `hourly_agg` workload, demonstrated a boost of up to 50% in performance.
```

Comment:
"such as those for the `hourly_agg` operation on the `http_logs` workload" (`hourly_agg` is not a workload)?

Reply:
@getsaurabh02 Could you confirm that we can make this change?

Reply:
looks good to me, OR "such as those for the `http_logs` workload"
yes - hourly_agg is not a workload)